### ISSUE TITLE
It seems that this code doesn't use negative triples to train embedding models.

### ISSUE THREAD

**Issue by davidlvxin** [salesforce/MultiHopKG#3]
Do this code use negative triples when training embedding models like DistMult? For example, given a triple (h, r, t), we need to create a negative triple (h', r, t') to use a margin-based ranking loss. I can't find codes related to them. Could you please help me to point them out? Thank you very much!

**Comment by todpole3** [comment]
This code does not use negative triples. We use the [BCELoss](https://github.com/salesforce/MultiHopKG/blob/master/src/emb/emb.py#26) which compute the loss for each (h, r) query considering all positive examples at once. 

For example, suppose there are four entities in the KG: A, B, C, D. If the training set contains triples 
(A, B, C)
(A, B, D),
the ground truth vector feed into the BCELoss is 
(0, 0, 1, 1). 

In this way we treat all other entities as negative examples. To avoid over penalizing false negatives as well as regularize the network, we use the [label smoothing technique](https://github.com/salesforce/MultiHopKG/blob/master/src/emb/emb.py#49).

The implementation is the same as https://github.com/TimDettmers/ConvE

**Comment by davidlvxin** [comment]
Thanks for your response. I have another question. Compared with margin-based ranking loss, could embedding models (DIstMult etc.) using BCELoss achieves an equivalent or better performance? Thank you very much!

**Comment by todpole3** [comment]
We did not compare head-to-head with margin-based ranking loss. My conjecture is that the performances of these two losses will be similar, but it's probably worth checking with experiments.

**Comment by davidlvxin** [comment]
OK, I know. Thanks a lot.
